librepo/yum.c

```diff
@@ -476,6 +476,8 @@ lr_check_repomd_xml_asc_availability(LrHandle *handle,
             g_clear_error(&tmp_err);
             unlink(signature);
             lr_free(signature);
+            lr_free(path);
+            close(fd);
```

Comment:
This doesn't seem right. Neither `fd` nor `path` is allocated/opened in this function, `fd` is not even used in this function besides closing it. This goes against common conventions of responsibility for deallocating. Whatever is passing these values to these functions might still be using them after this is called.

Having a look from where `lr_check_repomd_xml_asc_availability()` is called, one place is here: https://github.com/rpm-software-management/librepo/blob/e422278e4f0c505334aa109b1e1eb4ab7398ac3b/librepo/metadata_downloader.c#L341

There it's passing a `GSList` items as both `path` and `fd`. Unless I'm missing something, you are leaving a dangling pointer by freeing the list item this way. Having a quick look at the code it looks like a huge mess. The lists seem to be never freed, so it leaks as well. I think it needs to be fixed too...

For the other call to `lr_check_repomd_xml_asc_availability()` at: https://github.com/rpm-software-management/librepo/blob/e422278e4f0c505334aa109b1e1eb4ab7398ac3b/librepo/yum.c#L1312

Both variables should be freed (closed for `fd`) inside the fail-check if.

Reply:
You are totally right. I tried to fix the leak with a minimal change of code and logic. I was also surprised by freeing allocated variable in other function... So I'll try to come up with a better fix.